### ISSUE TITLE
force option to LibGit2.push is confusing

### ISSUE THREAD

**Issue by Keno** [JuliaLang/julia#23057]
In that I expected it to do something. Instead, it's ignored and the proper way to force push is to prepend a `+` to your refspec.

**Comment by kshyatt** [comment]
This method is insanely problematic, it seems. Do you want a doc fix here, or should we try to actually do what the option claims?

**Comment by StefanKarpinski** [comment]
Making it do what it should do would be preferable. Just because libgit2 has bad APIs doesn't mean we need to.

**Comment by Keno** [comment]
I don't think the option works with the current API that takes explicit refspecs. If there was a higher level api that built the refspec for you, I think having the option would be nice.

**Comment by StefanKarpinski** [comment]
So maybe just delete the option entirely since it doesn't do anything? Either that or use it's value to add or remove the `+` on the refspec as necessary (if it's not given, do nothing?).

**Comment by Keno** [comment]
Yes, for now remove entirely and update the documentation to tell you how to do it.

**Comment by staticfloat** [comment]
I just ran into this; it would be really nice to get rid of the fake `force` option.
